solr/core/src/java/org/apache/solr/util/stats/MetricUtils.java

```diff
@@ -165,6 +169,71 @@ public static void toSolrInputDocuments(
         });
   }
 
+  /**
+   * Provides a representation of the given Dropwizard metric registry as {@link
+   * SolrPrometheusCoreExporter}-s. Only those metrics are converted which match at least one of the
+   * given MetricFilter instances.
+   *
+   * @param registry the {@link MetricRegistry} to be converted
+   * @param shouldMatchFilters a list of {@link MetricFilter} instances. A metric must match <em>any
+   *     one</em> of the filters from this list to be included in the output
+   * @param mustMatchFilter a {@link MetricFilter}. A metric <em>must</em> match this filter to be
+   *     included in the output.
+   * @param propertyFilter limit what properties of a metric are returned
+   * @param skipHistograms discard any {@link Histogram}-s and histogram parts of {@link Timer}-s.
+   * @param skipAggregateValues discard internal values of {@link AggregateMetric}-s.
+   * @param compact use compact representation for counters and gauges.
+   * @param consumer consumer that accepts produced {@link SolrPrometheusCoreExporter}-s
+   */
+  public static void toPrometheusRegistry(
+      MetricRegistry registry,
+      String registryName,
+      List<MetricFilter> shouldMatchFilters,
+      MetricFilter mustMatchFilter,
+      Predicate<CharSequence> propertyFilter,
+      boolean skipHistograms,
+      boolean skipAggregateValues,
+      boolean compact,
+      Consumer<SolrPrometheusCoreExporter> consumer) {
+    String coreName;
+    boolean cloudMode = false;
+    Map<String, Metric> dropwizardMetrics = registry.getMetrics();
+    String[] rawParsedRegistry = registryName.split("\\.");
+    List<String> parsedRegistry = new ArrayList<>(Arrays.asList(rawParsedRegistry));
+
+    if (parsedRegistry.size() == 3) {
+      coreName = parsedRegistry.get(2);
+    } else if (parsedRegistry.size() == 5) {
+      coreName = parsedRegistry.stream().skip(1).collect(Collectors.joining("_"));
+      cloudMode = true;
+    } else {
+      coreName = registryName;
+    }
+
+    SolrPrometheusCoreExporter solrPrometheusCoreExporter =
+        new SolrPrometheusCoreExporter(coreName, cloudMode);
+
+    toMaps(
+        registry,
+        shouldMatchFilters,
+        mustMatchFilter,
+        propertyFilter,
+        skipHistograms,
+        skipAggregateValues,
+        compact,
+        false,
+        (metricName, metric) -> {
+          try {
+            Metric dropwizardMetric = dropwizardMetrics.get(metricName);
```

Comment:
A bit confused where you would prefer this. The metric is retrieved within the scope of the `BiConsumer` from `toMap`. I can't move it where `SolrPrometheusCoreExporter` is declared because it is out of scope of this block and the metrics are looped through the registry from `toMaps`.